packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -132,6 +136,36 @@ class ImageCache {
     return false;
   }
 
+  /// Returns a [Future] that waits for all currently pending images to finish
+  /// loading at least one frame.
+  ///
+  /// In a Flutter widget test, this must be used with [WidgetTester.runAsync]
+  /// or [TestWidgetsFlutterBinding.runAsync].
+  @visibleForTesting
+  Future<void> waitForPendingImages() {
+    final List<Future<void>> waitList = <Future<void>>[];
+    for (final _PendingImage pendingImage in _pendingImages.values) {
+      final Completer<void> completer = Completer<void>();
```

Comment:
Is there a way to consume a timeout here?

We could throw an exception with the list of images we were trying to load which would give a good indication to the user on why their test is not completing.